src/openrct2/actions/GameAction.cpp

```diff
@@ -197,12 +197,13 @@ namespace GameActions
             {
                 if (network_get_mode() == NETWORK_MODE_SERVER && result->Error == GA_ERROR::OK)
                 {
-                    uint32_t playerId = action->GetPlayer();
+                    const uint32_t playerId = action->GetPlayer();
+                    const uint32_t playerIndex = network_get_player_index(playerId);
 
-                    network_set_player_last_action(network_get_player_index(playerId), action->GetType());
+                    network_set_player_last_action(playerIndex, action->GetType());
                     if (result->Cost != 0)
                     {
-                        network_add_player_money_spent(playerId, result->Cost);
+                        network_add_player_money_spent(playerIndex, result->Cost);
```

Comment:
Given how easy it is to confuse the two:
a) why do we even have two of them in the first place?
b) they should have incompatible types, so they cannot be such easily confused.

Reply:
The game action logging PR introduces new types for this stuff, it needs some cleaning afterwards to adopt everything later to the new game action system rather than the old game command enums and types.